rocAL/include/decoders/video/rocdec_video_decoder.h

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include "video_decoder.h"
+#include "video_demuxer.h"
+#include "video_post_process.h"
+#include "rocvideodecode/roc_video_dec.h"
+
+#ifdef ROCAL_VIDEO
+#ifdef ENABLE_HIP
+
+typedef enum ReconfigFlushMode_enum {
```

Comment:
ReconfigFlushMode_enum - check if should consider renaming this enum.?
Why does it have to end with _enum ?

Reply:
If enum is needed, can we make it ReconfigFlushModeEnum ? Instead of combing the snake case and camel_case